### PR TITLE
AWS Terraform doc updates

### PR DESCRIPTION
**Pull request by stevenGravy**
- The **node** label was not clear what its usage was.
- no link to the AWS Terraform guide was given from the aws ha terraform README

### CODE REVIEW

**Review by webvictim — APPROVED** [review]

File: examples/aws/terraform/ha-autoscale-cluster/README.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ We recommend familiarizing yourself with the following resources prior to review
 
 - [Teleport Architecture](https://goteleport.com/docs/architecture/overview/)
 - [Admin Guide](https://goteleport.com/docs/management/admin/)
+- AWS HA Teleport Terraform Guide: [Running Teleport Enterprise in High Availability mode on AWS](https://goteleport.com/docs/deploy-a-cluster/deployments/aws-terraform/)
```
Comment (alexfornuto) [inline examples/aws/terraform/ha-autoscale-cluster/README.md:25]: Sort of duplicates the title, eh? Also aligns with existing format.

```suggestion
- [Running Teleport Enterprise in High Availability mode on AWS](https://goteleport.com/docs/deploy-a-cluster/deployments/aws-terraform/)
```

File: docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx [hunk 2]
```diff
@@ -295,12 +295,13 @@ $ terraform import aws_acm_certificate.cert <certificate_arn>
...
 
-Our reference deployment will provision the following instances for your cluster by default:
+Our reference deployment will provision the following instances for your cluster by default with AWS Auto Scaling Groups (ASG):
 
 - 2 x m4.large Teleport **auth** instances in an ASG, behind an internal network load balancer, configured using DynamoDB for
   shared storage. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/ha-autoscale-cluster/auth_asg.tf#L11)
 - 2 x m4.large Teleport **proxy** instances in an ASG, behind a public-facing load balancer - NLB for LetsEncrypt, ALB for ACM. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/ha-autoscale-cluster/proxy_asg.tf#L12)
-- 1 x m4.large Teleport **node** instance in an ASG. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/ha-autoscale-cluster/node_asg.tf#L10)
+- 1 x m4.large Teleport **node** instance in an ASG. Server instances in this ASG will join with Server SSH Access enabled connected to the **auth**
```
Comment (alexfornuto) [inline docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx:303]:
```suggestion
- 1 x m4.large Teleport **node** instance in an ASG. Server instances in this ASG will join with Server SSH Access enabled and connected to the **auth**
```

**Review by alexfornuto — CHANGES_REQUESTED** [review]
Couple of nits. Used this mode since auto-merge is enabled.

**Comment by stevenGravy** [comment]
> Couple of nits. Used this mode since auto-merge is enabled.

applied @alexfornuto , thanks!

**Review by alexfornuto — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

File: docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx [hunk 3]
```diff
@@ -295,12 +295,13 @@ $ terraform import aws_acm_certificate.cert <certificate_arn>
 
 ### Instances
 
-Our reference deployment will provision the following instances for your cluster by default:
+Our reference deployment will provision the following instances for your cluster by default with AWS Auto Scaling Groups (ASG):
```
Comment (zmb3) [inline docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx:298]: It's no longer clear if the _by default_ refers to ASGs for provisioning instances.

File: docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx [hunk 4]
```diff
@@ -295,12 +295,13 @@ $ terraform import aws_acm_certificate.cert <certificate_arn>
...
 
 - 2 x m4.large Teleport **auth** instances in an ASG, behind an internal network load balancer, configured using DynamoDB for
   shared storage. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/ha-autoscale-cluster/auth_asg.tf#L11)
 - 2 x m4.large Teleport **proxy** instances in an ASG, behind a public-facing load balancer - NLB for LetsEncrypt, ALB for ACM. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/ha-autoscale-cluster/proxy_asg.tf#L12)
-- 1 x m4.large Teleport **node** instance in an ASG. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/ha-autoscale-cluster/node_asg.tf#L10)
+- 1 x m4.large Teleport **node** instance in an ASG. Server instances in this ASG will join with Server SSH Access enabled and connected to the **auth**
```
Comment (zmb3) [inline docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx:303]: I'm not sure I've seen us use the term _Server SSH Access_ before. Don't we usually just say _Server Access_ or _with Teleport's SSH service enabled_?

File: docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx (hunk 3 above)
Comment (stevenGravy) [reply docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx]: Update language to indicate settings.  This also referred to all the settings like server type and count.

File: docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx (hunk 4 above)
Comment (stevenGravy) [reply docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx]: update to Teleport SSH Service

File: docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx [hunk 5]
```diff
@@ -295,12 +295,13 @@ $ terraform import aws_acm_certificate.cert <certificate_arn>
...
+Our reference deployment will provision the following instances for your cluster using AWS Auto Scaling Groups (ASG) with these default settings:
 
 - 2 x m4.large Teleport **auth** instances in an ASG, behind an internal network load balancer, configured using DynamoDB for
   shared storage. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/ha-autoscale-cluster/auth_asg.tf#L11)
 - 2 x m4.large Teleport **proxy** instances in an ASG, behind a public-facing load balancer - NLB for LetsEncrypt, ALB for ACM. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/ha-autoscale-cluster/proxy_asg.tf#L12)
-- 1 x m4.large Teleport **node** instance in an ASG. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/ha-autoscale-cluster/node_asg.tf#L10)
+- 1 x m4.large Teleport **node** instance in an ASG. Server instances in this ASG will join with Teleport SSH Service enabled and connected to the **auth**
```
Comment (zmb3) [inline docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx:303]: Are the nodes connecting directly to auth or via the proxy?

Comment (stevenGravy) [reply docs/pages/deploy-a-cluster/deployments/aws-terraform.mdx:303]: To the auth

**Review by zmb3 — APPROVED** [review]

**Pull request merged**